eva_cttv_pipeline/evidence_string_generation/clinvar_to_evidence_strings.py

```diff
@@ -145,101 +134,96 @@ def validate_evidence_string(ev_string, clinvar_record, trait, ensembl_gene_id,
+def clinvar_to_evidence_strings(string_to_efo_mappings, variant_to_gene_mappings, clinvar_xml, ot_schema,
+                                output_evidence_strings):
+    report = Report(trait_mappings=string_to_efo_mappings)
     ot_schema_contents = json.loads(open(ot_schema).read())
-    output_evidence_strings_file = utilities.open_file(output_evidence_strings, 'wt')
-    for cellbase_record in cell_recs:
-        report.counters["record_counter"] += 1
-        if report.counters["record_counter"] % 1000 == 0:
-            logger.info("{} records processed".format(report.counters["record_counter"]))
+    output_evidence_strings_file = file_utils.open_file(output_evidence_strings, 'wt')
 
+    logger.info('Processing ClinVar records')
+    for clinvar_record in clinvar_xml_utils.ClinVarDataset(clinvar_xml):
+        report.counters['record_counter'] += 1
+        if report.counters['record_counter'] % 1000 == 0:
+            logger.info('{} records processed'.format(report.counters['record_counter']))
         n_ev_strings_per_record = 0
-        clinvar_record = clinvar.ClinvarRecord(cellbase_record['clinvarSet'])
-
-        for clinvar_record_measure in clinvar_record.measures:
-            report.counters["n_nsvs"] += (clinvar_record_measure.nsv_id is not None)
-            append_nsv(report.nsv_list, clinvar_record_measure)
-            report.counters["n_multiple_allele_origin"] += (len(clinvar_record.allele_origins) > 1)
-            traits = create_traits(clinvar_record.traits, mappings.trait_2_efo, report)
-            converted_allele_origins = convert_allele_origins(clinvar_record.allele_origins)
-
-            for consequence_type, trait, allele_origin in itertools.product(
-                    get_consequence_types(clinvar_record_measure, mappings.consequence_type_dict),
-                    traits,
-                    converted_allele_origins):
 
-                if consequence_type is None:
-                    report.counters["no_variant_to_ensg_mapping"] += 1
-                    continue
+        if clinvar_record.measure is None:  # No valid variants to process in this record
+            continue
+        report.counters['n_nsvs'] += (clinvar_record.measure.nsv_id is not None)
+        append_nsv(report.nsv_list, clinvar_record.measure)
+        report.counters['n_multiple_allele_origin'] += (len(clinvar_record.allele_origins) > 1)
+        converted_allele_origins = convert_allele_origins(clinvar_record.allele_origins)
+        for consequence_type, clinvar_trait, allele_origin in itertools.product(
+                get_consequence_types(clinvar_record.measure, variant_to_gene_mappings),
+                clinvar_record.traits,
+                converted_allele_origins):
+
+            # Skip records for which crucial information (consequences or EFO mappings) is not available
+            if consequence_type is None:
+                report.counters['no_variant_to_ensg_mapping'] += 1
```

Comment:
To be consistent with the naming convention above, shouldn't this be "n_no_variant_to_ensg_mapping"?

Reply:
The whole reporting system/quality control system is super old and will also be refactored at some point (https://github.com/EBIvariation/eva-opentargets/issues/114). But I agree that consistency is important, so fixed this one: https://github.com/EBIvariation/eva-opentargets/pull/167/commits/7b455c2a2b53219920b7c691eabe2c3d3971ff2c